### ISSUE TITLE
h5py minimum version update?

### ISSUE THREAD

**Issue by hmaarrfk** [h5netcdf/h5netcdf#191]
Is there a minimum version for h5py that you are targeting?

It seems you are still targetting 2.10? is there a reason for this:
https://github.com/h5netcdf/h5netcdf/blob/main/.github/workflows/main.yml#L59

It seems 2.10 was released in 2019. Is that how far back you want to go?
https://pypi.org/project/h5py/2.10.0/

**Comment by hmaarrfk** [comment]
FYI, according to `nep29` as far back as 3.2 should be supported:
```
$ nep29 h5py
| version |    date    |
|:-------:|:----------:|
|  3.7.0  | 2022-05-24 |
|  3.6.0  | 2021-11-16 |
|  3.5.0  | 2021-10-20 |
|  3.4.0  | 2021-08-23 |
|  3.3.0  | 2021-06-22 |
|  3.2.0  | 2021-03-03 |
```

**Comment by kmuehlbauer** [comment]
@hmaarrfk I was thinking when to drop h5py2 in https://github.com/h5netcdf/h5netcdf/issues/176#issuecomment-1324646559. But maybe I'm overly concerned here and we should drop anyway? 

Maybe @shoyer has some guidance here?

**Comment by shoyer** [comment]
I usually would aim to support releases from the last 1-2 years worth of releases for an upstream dependency, depending upon how stable the upstream project is and how much work it takes to support.

So I would definitely drop h5py 2.x at this point.

**Comment by kmuehlbauer** [comment]
Great, thanks @shoyer! So we can get rid of the compatibility code and streamline the CI. 

That's best done in it's own PR before the performance PR's kick in. I can take care of this later today.

**Comment by kmuehlbauer** [comment]
Resolved by #194, thanks @shoyer and @hmaarrfk!
